Adding bicep file for webapp managed mysql quickstart

juliakm opened the pull request:
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)


- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Adding a bicep deploy file for the [Build a Web App on Linux with Azure database for MySQL sample](https://github.com/Azure/azure-quickstart-templates/tree/master/quickstarts/microsoft.web/webapp-linux-managed-mysql). I'd like to use reference this update in this [existing quickstart for Azure DevOps Pipelines](https://docs.microsoft.com/azure/devops/pipelines/apps/cd/azure/deploy-arm-template?view=azure-devops).

azure-quickstarts commented:
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

azure-quickstarts commented:
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

juliakm commented:
#sign-off

jbpaux commented on quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep:
```diff
@@ -0,0 +1,146 @@
+
+resource serverName_databaseName 'Microsoft.DBforMySQL/servers/databases@2017-12-01' = {
+  parent: serverName
+  name: '${databaseName}'
```
interpolation is not needed

jbpaux commented on quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep:
```diff
@@ -0,0 +1,146 @@
+}
+
+resource serverName 'Microsoft.DBforMySQL/servers@2017-12-01' = {
+  name: serverName_var
```
resource should be something like `mysqlserver` and the `name` attribute be `serverName` (was caused by decompile but should be cleaned afterwards)

jbpaux commented on quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep:
```diff
@@ -0,0 +1,146 @@
+  }
+}
+
+resource siteName_resource 'Microsoft.Web/sites@2020-06-01' = {
```
resource should be something like `site` (was caused by decompile but should be cleaned afterwards)

jbpaux commented on quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep:
```diff
@@ -0,0 +1,146 @@
+
+var databaseName = 'database${uniqueString(resourceGroup().id)}'
+var serverName_var = 'mysql-${uniqueString(resourceGroup().id)}'
+var hostingPlanName_var = 'hpn-${uniqueString(resourceGroup().id)}'
```
Rename variables to omit `_var`

jbpaux commented on quickstarts/microsoft.web/webapp-linux-managed-mysql/azuredeploy.bicep:
```diff
@@ -0,0 +1,146 @@
+var hostingPlanName_var = 'hpn-${uniqueString(resourceGroup().id)}'
+
+resource hostingPlanName 'Microsoft.Web/serverfarms@2020-06-01' = {
+  name: hostingPlanName_var
```
resource should be something like `hostingPlan` and the `name` attribute be `hostingPlanName` (was caused by decompile but should be cleaned afterwards)

jbpaux reviewed (changes requested):
I added a few comments here and you must also compile the bicep file

jbpaux commented:
You need to compile the bicep to azuredeploy.json and commit the file.

azure-quickstarts commented:
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

azure-quickstarts commented:
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

azure-quickstarts commented:
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

juliakm commented:
@bmoore-msft I think I've addressed the feedback and pipelines errors. Is there an example I can look at for a template without bpa errors? I'm not sure how to fix those.

azure-quickstarts commented:
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

azure-quickstarts commented:
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

juliakm commented:
@bmoore-msft Can you take a look at this again?

bmoore-msft commented on quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep:
```diff
@@ -0,0 +1,149 @@
+@description('main.bicep')
```
fix description

bmoore-msft commented on quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep:
```diff
@@ -0,0 +1,149 @@
+@description('main.bicep')
+param siteName string = '${uniqueString(resourceGroup().name)}'
```
remove string interpolation and use resourceGroup().id as the seed

```param siteName string = uniqueString(resourceGroup().id)```

bmoore-msft commented on quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep:
```diff
@@ -0,0 +1,149 @@
+@description('Azure database for mySQL sku family')
+param databaseskuFamily string = 'Gen5'
+
+var databaseName = '${uniqueString(resourceGroup().id)}'
```
in the original sample, these 3 vars (and the siteName param) had static prefixes to make them easier to read - is removing that intentional?  If so, remove the string interpolation on all 3...  If not, add back the prefixes here and on the site name param

juliakm replied:
It was in response to a different comment. I've removed the string interpolation on all three.

bmoore-msft commented on quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep:
```diff
@@ -0,0 +1,149 @@
+    serverFarmId: hostingPlan.id
+    httpsOnly: true
+  }     
+   identity: {
```
See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties for the proper sort order of properties

juliakm replied:
I think that this section was the only one with the incorrect order. Should be resolved now.

bmoore-msft commented on quickstarts/microsoft.web/webapp-linux-managed-mysql/main.bicep:
```diff
@@ -0,0 +1,149 @@
+    startIpAddress: '0.0.0.0'
+    endIpAddress: '255.255.255.255'
+  }
+  dependsOn: [
```
remove dependsOn

juliakm replied:
Removed now.

azure-quickstarts commented:
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

azure-quickstarts commented:
@juliakm - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]
